core/Measurement.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Diagnostics;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CTP.core
+{
+    public sealed class Measurements
+    {
+        private DataTable _table = new();
+        public double MaxValue = 10;
+        public double MinValue = 0;
+        public Measurements() { }
+
+        private static Measurements? _instance;
+
+        public static Measurements GetInstance()
+        {
+            _instance ??= new Measurements();
+            return _instance;
+        }
+
+        public void SetDataTable(DataTable Table)
+        {
+            _table = Table;
+
+            foreach (DataRow Row in _table.Rows)
+            {
+                double value = Row.Field<double>(1);
+                if (value > MaxValue) MaxValue = value;
+                else if(value < MinValue) MinValue = value;
+            }
+        }
+
+        public double? ReadValue(int index, int col = 1)
+        {
+            if (_table == null || index >= ValueCount()) return null;
```

Comment:
Bardzo dużo wywołań ValueCount(), gdyby to był atrybut to byłoby szybciej.

Reply:
Tak mi się wydaje, że musielibyśmy wtedy przenieść logikę do sprawdzania czy tabela nie jest pusta i czy nie wychodzimy poza zakres do każdego wykresu.

Reply:
Czemu?

Reply:
No w sensie przy odwołaniu do nieistniejącego elementu będziemy mieli exception, a na samym początku tablica jest nullem.

Dodatkowo teraz mamy odseparowaną logikę pobierania danych od wykresu, możemy np. zmienić typ danych w Measurement.cs, uwzględnić zmiany w funkcji ReadValue i nic nie będziemy musieli zmieniać w klasie RealTimeChart.